packages/flutter/lib/src/material/slider.dart

```diff
@@ -562,149 +610,197 @@ class _RenderSlider extends RenderBox {
+      constraints.hasBoundedHeight ? constraints.maxHeight : _overlayDiameter,
     );
   }
 
+  void _paintTickMarks(
+      Canvas canvas, Rect railLeft, Rect railRight, Paint leftPaint, Paint rightPaint) {
+    if (isDiscrete) {
+      // The ticks are tiny circles that are the same height as the rail.
+      const double tickRadius = _railHeight / 2.0;
+      final double railWidth = railRight.right - railLeft.left;
+      final double dx = (railWidth - _railHeight) / divisions;
+      // If the ticks would be too dense, don't bother painting them.
+      if (dx >= 3.0 * _railHeight) {
+        for (int i = 0; i <= divisions; i += 1) {
+          final double left = railLeft.left + i * dx;
+          final Offset center = new Offset(left + tickRadius, railLeft.top + tickRadius);
+          if (railLeft.contains(center)) {
+            canvas.drawCircle(center, tickRadius, leftPaint);
+          } else if (railRight.contains(center)) {
+            canvas.drawCircle(center, tickRadius, rightPaint);
+          }
+        }
+      }
+    }
+  }
+
+  void _paintOverlay(Canvas canvas, Offset center) {
+    if (!_reaction.isDismissed) {
+      // TODO(gspencer) : We don't really follow the spec here for overlays.
+      // The spec says to use 16% opacity for drawing over light material,
+      // and 32% for colored material, but we don't really have a way to
+      // know what the underlying color is, so there's no easy way to
+      // implement this. Choosing the "light" version for now.
+      final Paint reactionPaint = new Paint()..color = _sliderTheme.overlayColor;
+      final double radius = _overlayRadiusTween.evaluate(_reaction);
+      canvas.drawCircle(center, radius, reactionPaint);
+    }
+  }
+
   @override
   void paint(PaintingContext context, Offset offset) {
     final Canvas canvas = context.canvas;
 
-    final double trackLength = size.width - 2 * _kReactionRadius;
-    final bool enabled = isInteractive;
-    final double value = _position.value;
-    final bool thumbAtMin = value == 0.0;
-
-    final Paint primaryPaint = new Paint()..color = enabled ? _activeColor : _inactiveColor;
-    final Paint trackPaint = new Paint()..color = _inactiveColor;
+    final double railLength = size.width - 2 * _overlayRadius;
+    final double value = _state.positionController.value;
+    final ColorTween activeRailEnableColor = new ColorTween(
+        begin: _sliderTheme.disabledActiveRailColor, end: _sliderTheme.activeRailColor);
+    final ColorTween inactiveRailEnableColor = new ColorTween(
+        begin: _sliderTheme.disabledInactiveRailColor, end: _sliderTheme.inactiveRailColor);
+    final ColorTween activeTickMarkEnableColor = new ColorTween(
+        begin: _sliderTheme.disabledActiveTickMarkColor, end: _sliderTheme.activeTickMarkColor);
+    final ColorTween inactiveTickMarkEnableColor = new ColorTween(
+        begin: _sliderTheme.disabledInactiveTickMarkColor, end: _sliderTheme.inactiveTickMarkColor);
```

Comment:
these aren't really wrapped in the flutter style either, though here i don't think it impacts readabilty much